### PR TITLE
add absolute links for source locations

### LINKED ISSUE #7960: Support absolute web links (at least) in PO files
### Describe the problem

Our translators are working on a documentation site that gets updated (in a staging area) almost instantly after they updated their translation. They have no intuitive, direct way to access the result of this rendering, or maybe even the situation _before_ the translation to get more context.

### Describe the solution you'd like

We are thinking about adding the absolute link as a reference (`#:`) to the PO files. They are then showing up as expected in the "Source String Location" section. The only issue is that weblate tries to interpret them as a VCS link, interpreting "https" as the file name and "//www.example.com/path/doc.html" as the line number.
If we could add a detection for web links (either checking for `https?://` or, more generically, for `[\w]+://`), which would lead to just linking to that absolute location, we could very seamlessly use that feature.

While this scenario is not mentioned in any documentation I could find about the "formal" gettext format, I do not think that it contradicts the intention of the reference field.

### Describe alternatives you've considered

Screenshots: Adding screenshots to source strings is cumbersome and does not provide the best experience for the translator. 

I don't know if there are alternative approaches, short of buliding some kind of add-on, hook, or the like.

### Screenshots

image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by oliverrahner**
## Proposed changes

Implements the change proposed in #7960 (closes #7960)

## Checklist

- [X] Lint and unit tests pass locally with my changes.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [X] I have squashed my commits into logic units.
- [X] I have described the changes in the commit messages.

### CODE REVIEW

File: weblate/trans/templatetags/translations.py [hunk 1]
```diff
@@ -603,9 +603,16 @@ def unit_state_title(unit) -> str:
...
-    """Attemp to convert `text` to a repo link to `filename:line`."""
+    """
+    Attempt to convert `text` to a repo link to `filename:line`.
+
+    If the `text` is prefixed with a protocol, like https://, http://, ftp://, the
+    link will be an absolute link to the specified resource.
+    """
     link = None
-    if profile:
+    if re.search(r"\w+://", text):
```
Comment (nijel) [inline weblate/trans/templatetags/translations.py:613]: I don't think we want to whitelist all protocols here. I'd stick with `http://` and `https://` only.

Comment (oliverrahner) [reply weblate/trans/templatetags/translations.py]: Agreed and changed!

File: weblate/trans/templatetags/translations.py [hunk 2]
```diff
@@ -606,11 +606,11 @@ def try_linkify_filename(
     """
     Attempt to convert `text` to a repo link to `filename:line`.
 
-    If the `text` is prefixed with a protocol, like https://, http://, ftp://, the
+    If the `text` is prefixed with http:// or https://, the
     link will be an absolute link to the specified resource.
     """
     link = None
-    if re.search(r"\w+://", text):
+    if re.search(r"https?://", text):
```
Comment (nijel) [inline weblate/trans/templatetags/translations.py:613]: Shouldn't it look only at string beginning?
```suggestion
    if re.match("^https?://", text):
```

Comment (oliverrahner) [reply weblate/trans/templatetags/translations.py]: fixed it, but re-added the raw string literal

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**